packages/dds/tree/src/test/shared-tree/sharedTreeFuzzTests.ts

```diff
@@ -0,0 +1,273 @@
+                const { testTreeProvider } = state;
+                if (testTreeProvider === undefined) {
+                    fail("Attempted to synchronize with undefined testObjectProvider");
+                }
+                await testTreeProvider.ensureSynchronized();
+                return state;
+            },
+        },
+        initialState,
+        saveInfo,
+    );
+
+    await finalState.testTreeProvider.ensureSynchronized();
+    validateTree(provider.trees[0], [initialTreeState]);
+
+    if (validateAnchor) {
+        const expectedPath: UpPath = {
+            parent: {
+                parent: undefined,
+                parentIndex: 0,
+                parentField: rootFieldKeySymbol,
+            },
+            parentField: brand("foo"),
+            parentIndex: 1,
+        };
+        const anchorPath = tree.locate(firstAnchor);
+        assert(compareUpPaths(expectedPath, anchorPath));
+    }
+    return finalState;
+}
+
+function applyFuzzChange(
+    tree: ISharedTree,
+    contents: FuzzChange,
+    transactionResult: TransactionResult,
+): void {
+    switch (contents.fuzzType) {
+        case "insert":
+            tree.runTransaction((forest, editor) => {
+                const field = editor.sequenceField(contents.parent, contents.field);
+                field.insert(
+                    contents.index,
+                    singleTextCursor({ type: brand("Test"), value: contents.value }),
+                );
+                return transactionResult;
+            });
+            break;
+        case "delete":
+            // generated edit returns path as undefined if no node exists
+            if (contents.path !== undefined) {
+                const parent = contents.path?.parent;
+                const delField = contents.path?.parentField;
+                const parentIndex = contents.path?.parentIndex;
+                tree.runTransaction((forest, editor) => {
+                    const field = editor.sequenceField(parent, delField);
+                    field.delete(parentIndex, 1);
+                    return transactionResult;
+                });
+            }
+            break;
+        case "setPayload":
+            // generated edit returns path as undefined if no node exists
+            if (contents.path !== undefined) {
+                const path = contents.path;
+                tree.runTransaction((forest, editor) => {
+                    editor.setValue(path, contents.value);
+                    return transactionResult;
+                });
+            }
+            break;
+        default:
+            fail("Invalid edit.");
+    }
+}
+
+function runBatch(
+    opGenerator: () => AsyncGenerator<Operation, FuzzTestState>,
+    fuzzActions: (
+        generatorFactory: AsyncGenerator<Operation, FuzzTestState>,
+        seed: number,
+    ) => Promise<FuzzTestState>,
+    opsPerRun: number,
+    batchSize: number,
+    random: IRandom,
+    skipTests?: boolean,
```

Comment:
I don't think this function needs to worry about the tests being skipped or not. If the caller wants the tests to be skipped when it can do `describe.skip(...` on the describe block.